libraries/AP_Motors/AP_MotorsMulticopter.cpp

```diff
@@ -421,12 +428,16 @@ void AP_MotorsMulticopter::update_lift_max_from_batt_voltage()
     // contrain resting voltage estimate (resting voltage is actual voltage with sag removed based on current draw and resistance)
     _batt_voltage_resting_estimate = constrain_float(_batt_voltage_resting_estimate, _batt_voltage_min, _batt_voltage_max);
 
-    // filter at 0.5 Hz
-    float batt_voltage_filt = _batt_voltage_filt.apply(_batt_voltage_resting_estimate / _batt_voltage_max, _dt);
+    if (!has_option(MotorOptions::BATT_RAW_VOLTAGE)) {
+        // filter at 0.5 Hz
+        _batt_voltage_filt.apply(_batt_voltage_resting_estimate / _batt_voltage_max, _dt);
+    } else {
+        _batt_voltage_filt.reset(_batt_voltage_resting_estimate / _batt_voltage_max);
```

Comment:
I think this reset is a bit confusing. I would rather see the filter frequency as a param. Or set the filter frequency to zero in this case.

Reply:
Or maybe just a comment to say the reset is the same as no filter.